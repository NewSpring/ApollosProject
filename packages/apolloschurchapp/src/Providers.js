import React from 'react';
import { Providers } from '@apollosproject/ui-kit';
import ClientProvider from './client';

import theme from './theme'; // where theme looks like:

const NewSpringProviders = (props) => (
  <Providers themeInput={theme}>
    <ClientProvider {...props} />
  </Providers>
);

export default NewSpringProviders;
