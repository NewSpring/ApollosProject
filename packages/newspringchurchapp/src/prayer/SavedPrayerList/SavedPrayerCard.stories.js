import React from 'react';
import { storiesOf } from '@storybook/react-native';

import SavedPrayerCard from './SavedPrayerCard';

storiesOf('prayer/SavedPrayer/SavedPrayerCard', module).add('Example', () => (
  <SavedPrayerCard />
));
