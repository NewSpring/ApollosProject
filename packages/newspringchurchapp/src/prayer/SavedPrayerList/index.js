import React from 'react';
import { ScrollView, View } from 'react-native';
import { Query, Mutation } from 'react-apollo';
import { styled } from '@apollosproject/ui-kit';

import PrayerCard from '../PrayerCard';
import savedPrayerList from '../data/queries/getSavedPrayers';
import deleteSavedPrayer from '../data/mutations/deleteSavedPrayer';

const StyledView = styled(({ theme }) => ({
  marginTop: theme.sizing.baseUnit * 1.5,
  marginBottom: theme.sizing.baseUnit * 4,
}))(View);

const SavedPrayerList = () => (
  <ScrollView>
    <Query query={savedPrayerList} fetchPolicy="cache-and-network">
      {({ data: { savedPrayers = [] } = {} }) => (
        <Mutation
          mutation={deleteSavedPrayer}
          update={async (cache, { data: { unSavePrayer } }) => {
            const currentSavedPrayers = cache.readQuery({
              query: savedPrayerList,
            });
            const { id } = unSavePrayer;
            const newPrayersList = currentSavedPrayers.savedPrayers.filter(
              (prayer) => prayer.id !== id
            );
            await cache.writeQuery({
              query: savedPrayerList,
              data: { savedPrayers: newPrayersList },
            });
          }}
        >
          {(unsavePrayer) => (
            <StyledView>
              {savedPrayers
                .map((prayer) => (
                  <PrayerCard
                    avatarSource={prayer.person.photo.uri}
                    avatarSize={'medium'}
                    name={prayer.firstName}
                    campus={prayer.campus.name}
                    key={prayer.id}
                    created={prayer.enteredDateTime}
                    prayer={prayer.text}
                    showHelp
                    header
                    options={[
                      {
                        title: 'Remove Prayer',
                        method: async () => {
                          await unsavePrayer({
                            variables: {
                              nodeId: prayer.id,
                            },
                          });
                        },
                        destructive: true,
                      },
                    ]}
                  />
                ))
                .reverse()}
            </StyledView>
          )}
        </Mutation>
      )}
    </Query>
  </ScrollView>
);

export default SavedPrayerList;
