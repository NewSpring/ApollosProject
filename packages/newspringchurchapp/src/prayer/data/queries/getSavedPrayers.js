import gql from 'graphql-tag';

export default gql`
  query savedPrayers {
    savedPrayers {
      id
      campus {
        id
        name
      }
      firstName
      lastName
      text
      enteredDateTime
      person {
        photo {
          uri
        }
      }
      isAnonymous
    }
  }
`;
