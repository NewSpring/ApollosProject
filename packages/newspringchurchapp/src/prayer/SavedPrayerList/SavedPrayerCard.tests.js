import React from 'react';
import renderer from 'react-test-renderer';

import Providers from 'newspringchurchapp/src/Providers';

import SavedPrayerCard from './SavedPrayerCard';

describe('SavedPrayerCard', () => {
  it('renders a saved prayer', () => {
    const tree = renderer.create(
      <Providers>
        <SavedPrayerCard />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
