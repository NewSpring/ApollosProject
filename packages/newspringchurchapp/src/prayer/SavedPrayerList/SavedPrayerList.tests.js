import React from 'react';

import renderer from 'react-test-renderer';
import Providers from 'newspringchurchapp/src/Providers';
import getSavedPrayers from '../data/queries/getSavedPrayers';
import SavedPrayerList from '.';

const savedPrayers = [
  {
    id: 'PrayerRequest:b99bbd5002a0bf8d4c73e86afe9d4786',
    firstName: 'Dan',
    lastName: 'Edwards',
    text: 'Praying for an around the world trip to happen sometime in 2020.',
    enteredDateTime: '2019-05-02T09:00:00.000',
  },
  {
    id: 'PrayerRequest:b99bbd5002a0bf8d4c73e86afe9d4786',
    firstName: 'Isaac',
    lastName: 'Hardy',
    text: 'I could really use some new Youtube subscribers to my channel.',
    enteredDateTime: '2019-05-02T09:00:00.000',
  },
  {
    id: 'PrayerRequest:b99bbd5002a0bf8d4c73e86afe9d4786',
    firstName: 'Michael',
    lastName: 'Neeley',
    text: 'Guys, just make a decision already.',
    enteredDateTime: '2019-05-02T09:00:00.000',
  },
  {
    id: 'PrayerRequest:b99bbd5002a0bf8d4c73e86afe9d4786',
    firstName: 'Rich',
    lastName: 'Dubay',
    text: 'I just want to live in a cave.',
    enteredDateTime: '2019-05-02T09:00:00.000',
  },
];

const mock = {
  request: {
    query: getSavedPrayers,
  },
  response: {
    data: {
      savedPrayers,
    },
  },
};

describe('the SavedPrayerList component', () => {
  it('renders a list of saved prayers', async () => {
    const tree = renderer.create(
      <Providers mocks={[mock]}>
        <SavedPrayerList />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
