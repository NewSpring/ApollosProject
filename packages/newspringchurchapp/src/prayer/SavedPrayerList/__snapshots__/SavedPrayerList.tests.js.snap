// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the SavedPrayerList component renders a list of saved prayers 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <RCTScrollView
      nestedScrollEnabled={true}
    >
      <View>
        <View
          style={
            Object {
              "marginBottom": 64,
              "marginTop": 24,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
